File: code-recipes/php/src/easyship/GetScheduledPackageRecipe.php
<?php

namespace Src\easyship;

use SpApi\Api\easyship\v2022_03_23\EasyShipApi;
use Src\util\Recipe;

/**
 * Code Recipe to retrieve a scheduled EasyShip package
 * Steps:
 * 1. Initialize EasyShip API client
 * 2. Call getScheduledPackage operation
 * 3. Validate Scheduled Package Id
 */
class GetScheduledPackageRecipe extends Recipe
{
    private EasyShipApi $easyShipApi;
    private string $amazonOrderId;
    private string $marketplaceId;

    public function start(): void
    {
        $this->initializeEasyShipApi();
        $packageResponse = $this->getScheduledPackage();
        $this->validateScheduledPackageId($packageResponse);
        echo "✅ Successfully retrieved and validated scheduled package\n";
    }

    private function initializeEasyShipApi(): void
    {
        $this->easyShipApi = new EasyShipApi($this->config);
        $this->amazonOrderId = "902-3159121-1390916";
        $this->marketplaceId = "A1AM78C64UM0Y8";
        echo "EasyShip API client initialized\n";
    }

    private function getScheduledPackage()
    {
        $response = $this->easyShipApi->getScheduledPackage($this->amazonOrderId, $this->marketplaceId);
        echo "Scheduled package retrieved for order: {$this->amazonOrderId}\n";
        return $response;
    }

    private function validateScheduledPackageId($packageResponse): void
    {
        $responseOrderId = $packageResponse->getScheduledPackageId()->getAmazonOrderId();
        if ($this->amazonOrderId !== $responseOrderId) {
            throw new \Exception("Scheduled Package Id mismatch. Expected: {$this->amazonOrderId}, Got: {$responseOrderId}");
        }
        echo "✅ Scheduled Package Id validated: {$responseOrderId}\n";
    }
}

File: code-recipes/java/src/main/java/easyship/GetScheduledPackageRecipe.java
package easyship;

import software.amazon.spapi.ApiException;
import software.amazon.spapi.api.easyship.v2022_03_23.EasyShipApi;
import software.amazon.spapi.models.easyship.v2022_03_23.ModelPackage;
import util.Recipe;
import util.Constants;

import com.amazon.SellingPartnerAPIAA.LWAException;

/**
 * Code Recipe to retrieve a scheduled EasyShip package
 * Steps:
 * 1. Initialize EasyShip API client
 * 2. Call getScheduledPackage operation
 * 3. Validate Scheduled Package Id
 */
public class GetScheduledPackageRecipe extends Recipe {

    private EasyShipApi easyShipApi;
    private String amazonOrderId;
    private String marketplaceId;

    @Override
    protected void start() {
        initializeEasyShipApi();
        ModelPackage packageResponse = getScheduledPackage();
        validateScheduledPackageId(packageResponse);
        System.out.println("✅ Successfully retrieved and validated scheduled package");
    }

    private void initializeEasyShipApi() {
        easyShipApi = new EasyShipApi.Builder()
                .lwaAuthorizationCredentials(lwaCredentials)
                .endpoint(Constants.BACKEND_URL)
                .disableAccessTokenCache()
                .build();
        amazonOrderId = "902-3159121-1390916";
        marketplaceId = "A1AM78C64UM0Y8";
        System.out.println("EasyShip API client initialized");
    }

    private ModelPackage getScheduledPackage() {
        try {
            ModelPackage response = easyShipApi.getScheduledPackage(amazonOrderId, marketplaceId);
            System.out.println("Scheduled package retrieved for order: " + amazonOrderId);
            return response;
        } catch (ApiException | LWAException e) {
            throw new RuntimeException("Failed to get scheduled package", e);
        }
    }

    private void validateScheduledPackageId(ModelPackage packageResponse) {
        String responseOrderId = packageResponse.getScheduledPackageId().getAmazonOrderId();
        if (!amazonOrderId.equals(responseOrderId)) {
            throw new IllegalStateException("Scheduled Package Id mismatch. Expected: " + amazonOrderId + ", Got: " + responseOrderId);
        }
        System.out.println("✅ Scheduled Package Id validated: " + responseOrderId);
    }
}
